pkg/kubelet/pleg/generic.go

```diff
@@ -390,6 +390,14 @@ func (g *GenericPLEG) updateCache(pod *kubecontainer.Pod, pid types.UID) error {
 		// a pod status after network teardown, but the kubernetes API expects
 		// the completed pod's IP to be available after the pod is dead.
 		status.IPs = g.getPodIPs(pid, status)
+
+		// Log exit code when containers finished
+		for _, containerStatus := range status.ContainerStatuses {
+			if !containerStatus.FinishedAt.IsZero() {
```

Comment:
So if one container in Pod exited as expected while other keep running, this message will keep being reported over and over? Is this the best place to report exit code?

Reply:
Thanks for the comments. updateCache() is called when there is new event generated for the Pod, so when Pod exited, there won't be any new event for it, therefore it will be reported only once. 

There are other places to emit exit code like appending exit code in PLEG containerDied event, but this requires a change of event data structure, pass podStatus to kubelet instead of in pleg relist loop.  so it is more complex than current one.

Reply:
I meant a single container exited in he pod while others are running. Are there other situations `updateCache` will be called?

Reply:
I see. If a pod has multiple containers, it will log all exit containers repeatedly when there is a new event of the pod. 

I think there is a better place for adding the log exit code: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/pleg/generic.go#L267

We can move the log logic in there, and check container id to avoid logging repeatedly.  What's your thought ?